### PR TITLE
Include new conditions.reason as part of maxUnavailable support.

### PR DESCRIPTION
**Pull request by geetikakay**
Signed-off-by: Geetika Kapoor <gkapoor@redhat.com>

##### Short description:
This is the new reason recently added under conditions as part of maxunavailable support.
Below is example how it gets displayed.
Status:
  Conditions:
    Last Hearbeat Time:    2021-11-16T16:29:01Z
    Last Transition Time:  2021-11-16T16:29:01Z
    Message:               Max unavailable node limit reached
    Reason:                MaxUnavailableLimitReached
    Status:                True
    Type:                  Pending

### CODE REVIEW

File: ocp_resources/node_network_configuration_policy.py [hunk 1]
```diff
@@ -27,12 +27,14 @@ class Type:
...
             PROGRESSING = "Progressing"
             MATCHING = "Matching"
+            PENDING = "Pending"
 
         class Reason:
             CONFIGURING = "ConfigurationProgressing"
             SUCCESS = "SuccessfullyConfigured"
             FAILED = "FailedToConfigure"
             NO_MATCHING_NODE = "NoMatchingNode"
+            MAX_UNAVAILABLE_LIMIT = "MaxUnavailableLimitReached"
```
Comment (rnetser) [inline ocp_resources/node_network_configuration_policy.py:37]:
```suggestion
            MAX_UNAVAILABLE_LIMIT_REACHED = "MaxUnavailableLimitReached"
```

**Review by rnetser — CHANGES_REQUESTED** [review]

**Comment by geetikakay** [comment]
These changes are done based on [code](https://github.com/nmstate/kubernetes-nmstate):

supported types and reasons:
enactment types
	NodeNetworkConfigurationEnactmentConditionAvailable   ConditionType = "Available"
	NodeNetworkConfigurationEnactmentConditionFailing     ConditionType = "Failing"
	NodeNetworkConfigurationEnactmentConditionPending     ConditionType = "Pending"
	NodeNetworkConfigurationEnactmentConditionProgressing ConditionType = "Progressing"
	NodeNetworkConfigurationEnactmentConditionAborted     ConditionType = "Aborted"

Policy types are 
	NodeNetworkConfigurationPolicyConditionAvailable ConditionType = "Available"
	NodeNetworkConfigurationPolicyConditionDegraded  ConditionType = "Degraded"


api/shared/nodenetworkconfigurationenactment_types.go:	NodeNetworkConfigurationEnactmentConditionFailedToConfigure          ConditionReason = "FailedToConfigure"
api/shared/nodenetworkconfigurationenactment_types.go:	NodeNetworkConfigurationEnactmentConditionSuccessfullyConfigured     ConditionReason = "SuccessfullyConfigured"
api/shared/nodenetworkconfigurationenactment_types.go:	NodeNetworkConfigurationEnactmentConditionMaxUnavailableLimitReached ConditionReason = "MaxUnavailableLimitReached"
api/shared/nodenetworkconfigurationenactment_types.go:	NodeNetworkConfigurationEnactmentConditionConfigurationProgressing   ConditionReason = "ConfigurationProgressing"
api/shared/nodenetworkconfigurationenactment_types.go:	NodeNetworkConfigurationEnactmentConditionConfigurationAborted       ConditionReason = "ConfigurationAborted"
api/shared/nodenetworkconfigurationpolicy_types.go:	NodeNetworkConfigurationPolicyConditionFailedToConfigure           ConditionReason = "FailedToConfigure"
api/shared/nodenetworkconfigurationpolicy_types.go:	NodeNetworkConfigurationPolicyConditionSuccessfullyConfigured      ConditionReason = "SuccessfullyConfigured"
api/shared/nodenetworkconfigurationpolicy_types.go:	NodeNetworkConfigurationPolicyConditionConfigurationProgressing    ConditionReason = "ConfigurationProgressing"
api/shared/nodenetworkconfigurationpolicy_types.go:	NodeNetworkConfigurationPolicyConditionConfigurationNoMatchingNode ConditionReason = "NoMatchingNode"

File: ocp_resources/node_network_configuration_enactment.py [hunk 2]
```diff
@@ -3,3 +3,18 @@
...
+    class Conditions:
+        class Type:
+            FAILING = "Failing"
+            AVAILABLE = "Available"
+            PROGRESSING = "Progressing"
+            PENDING = "Pending"
+            ABORTED = "Aborted"
+
+        class Reason:
+            CONFIGURING = "ConfigurationProgressing"
```
Comment (myakove) [inline ocp_resources/node_network_configuration_enactment.py:16]: Please match the constant name to the actual reason.
CONFIGURATION_PROGERESSING = "ConfigurationProgressing"
and so on

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/node_network_configuration_enactment.py (hunk 2 above)
Comment (geetikakay) [reply ocp_resources/node_network_configuration_enactment.py:16]: this should be for both files(Enactment + policy) or just Enactment?

Comment (myakove) [reply ocp_resources/node_network_configuration_enactment.py:16]: for both, thx.

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocp_resources/node_network_configuration_enactment.py (hunk 2 above)
Comment (geetikakay) [reply ocp_resources/node_network_configuration_enactment.py]: Done

**Review by myakove — APPROVED** [review]

**Pull request merged**